---
title: "RNAseq analysis"
author: "Joseph A. Tandurella, Ludmila Danilova"
date: "`r Sys.Date()`"
output:
  html_document:
    toc: true
    toc_float: true
---

# R libraries and versions

Loading of R packages, setting seed for reproducability, and session info to provide version numbers.

```{r}
knitr::opts_chunk$set(echo = TRUE, message = FALSE,
                      warning = FALSE)
rm(list = ls())

library(tximport)
library('readxl')
library('DESeq2')
library('ComplexHeatmap')
library('EnhancedVolcano')
library('DT')
library('fgsea')
library(ggplot2)
library(ggpubr)
library(tidyverse)
library(dplyr)
library(tidyr)
library(data.table)
library(msigdbr)
library(fgsea)
library(Matrix)
library(ashr)

set.seed(1234)

# set root folder
rootDir = "../"

# load shared parameters and functions
source(paste0(rootDir,"parameters_and_functions.r"))

```


# Load expression data

```{r}

exprsDat = read.csv(paste0(rootDir,"input_data/table_RNAseq.csv"), row.names = 1)

dim(exprsDat)

```


# Sample Annotations and counts

```{r}
#=========================
# read in patient annotation and mapping
patTab = read.csv(file = paste0(rootDir,"input_data/table_patients.csv"), row.names = 1)
# switch to R/NR annotation
patTab = patTab %>% mutate(RECIST = factor(RECIST,levels = c("Responder", "NonResponder"), labels = c("R","NR")))

datatable(patTab)

# sample annotation
sampAnnot = data.frame(
  patientID = paste(sapply(strsplit(colnames(exprsDat), '_'), geti,1), sapply(strsplit(colnames(exprsDat), '_'), geti,2), sep = '_'),
  timepoint = sapply(strsplit(colnames(exprsDat), '_'), geti,3), 
  check.names = F)

rownames(sampAnnot) = colnames(exprsDat)

table(sampAnnot$timepoint)

# find pairs
mat = printPairs(sampAnnot)

```

# Data normalization and exploratory data analysis


## Evaluation of sample quality based upon distribution of reads

We evaluate sample quality from the distribution of reads as visualized in a boxplot of log counts. 

```{r}
boxplot(log2(exprsDat+1), ylab='log2(counts+1)', 
        xlab='samples', las=2, cex.axis = 0.25, 
        border = ifelse(apply(log2(exprsDat+1),2,median)==0,
                        'red','black'), main = "Distribution of log-transformed counts")

```

## Subset to paired samples

Due to this analysis being paired, samples with matching timepoints were retained across all comparisons. Samples without matching timepoints were removed. 

```{r}
##BaselinevC1D1
pairedsamples <- mat %>% filter(Baseline == 1 & C1D1 == 1) %>% rownames()
# select paired samples and exclude a time point from comparison
BaselinevC1D1sampAnnotFilter <- sampAnnot %>% filter(patientID %in% pairedsamples & timepoint != "C2D1")
# subset expression data
BaselinevC1D1exprsDatFilter <- exprsDat[,rownames(BaselinevC1D1sampAnnotFilter)]

##BaselinevC2D1
pairedsamples <- mat %>% filter(Baseline == 1 & C2D1 == 1) %>% rownames()
# select paired samples and exclude a time point from comparison
BaselinevC2D1sampAnnotFilter <-  sampAnnot %>% filter(patientID %in% pairedsamples & timepoint != "C1D1")
# subset expression data
BaselinevC2D1exprsDatFilter <- exprsDat[,rownames(BaselinevC2D1sampAnnotFilter)]

##C1D1vC2D1
pairedsamples <- mat %>% filter(C1D1 == 1 & C2D1 == 1) %>% rownames()
# select paired samples and exclude a time point from comparison
C1D1vC2D1sampAnnotFilter <- sampAnnot %>% filter(patientID %in% pairedsamples & timepoint != "Baseline")
# subset expression data
C1D1vC2D1exprsDatFilter <- exprsDat[,rownames(C1D1vC2D1sampAnnotFilter)]

##AllFinalSamples
sampAnnotPaired <- sampAnnot[rownames(sampAnnot) %in% c(rownames(BaselinevC1D1sampAnnotFilter),rownames(BaselinevC2D1sampAnnotFilter),rownames(C1D1vC2D1sampAnnotFilter)),]
exprsDatPaired <- exprsDat[,rownames(sampAnnotPaired)]

```


# Paired differential expression analysis by Time (Treatment)

For the retained samples, we run differential expression analysis across time with [DESeq2](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0550-8). Estimated fold changes are shrunk with `ashr` using `lfcShrink` to account for the variation in the samples in this dataset. 

```{r}

#DE
  #=======================
##Baseline V Entino
# get samples from baseline and C1D1
  samp <- BaselinevC1D1sampAnnotFilter[!is.na(BaselinevC1D1sampAnnotFilter$timepoint),]
  sampIds <- rownames(samp)
# create a DESeq object  
BaselinevC1D1dds <- DESeqDataSetFromMatrix(
    countData =                             round(BaselinevC1D1exprsDatFilter)[,colnames(BaselinevC1D1exprsDatFilter) %in% sampIds], 
    colData = samp, 
    design=~patientID+timepoint)

  BaselinevC1D1dds <- DESeq(BaselinevC1D1dds)
  
 # get the result table 
  BaselinevC1D1res <- results(BaselinevC1D1dds, contrast = c("timepoint","C1D1", "Baseline"))
  BaselinevC1D1res <- lfcShrink(BaselinevC1D1dds, contrast = c("timepoint","C1D1", "Baseline"), res=BaselinevC1D1res, type="ashr")
  BaselinevC1D1res <- BaselinevC1D1res[!is.na(BaselinevC1D1res$padj),]

  #=======================
##Baseline V C2D1
  samp <- BaselinevC2D1sampAnnotFilter[!is.na(BaselinevC2D1sampAnnotFilter$timepoint),]
  sampIds <- rownames(samp) 
  
  ##DE by Time
  BaselinevC2D1dds <- DESeqDataSetFromMatrix(countData =
                                  round(BaselinevC2D1exprsDatFilter)[,colnames(BaselinevC2D1exprsDatFilter) %in% sampIds], 
                              colData = samp, 
                              design=~patientID+timepoint)
  
  BaselinevC2D1dds <- DESeq(BaselinevC2D1dds)
  BaselinevC2D1res <- results(BaselinevC2D1dds, contrast = c("timepoint","C2D1", "Baseline"))
  BaselinevC2D1res <- lfcShrink(BaselinevC2D1dds, contrast = c("timepoint","C2D1", "Baseline"), res=BaselinevC2D1res, type="ashr")
  BaselinevC2D1res <- BaselinevC2D1res[!is.na(BaselinevC2D1res$padj),]

  #=======================
##Entino v C2D1
# DE 
  samp <- C1D1vC2D1sampAnnotFilter[!is.na(C1D1vC2D1sampAnnotFilter$timepoint),]
  sampIds <- rownames(samp)
  
  ##DE by Time
  C1D1vC2D1dds <- DESeqDataSetFromMatrix(countData =
                                  round(C1D1vC2D1exprsDatFilter)[,colnames(C1D1vC2D1exprsDatFilter) %in% sampIds], 
                              colData = samp, 
                              design=~patientID+timepoint)
  C1D1vC2D1dds <- DESeq(C1D1vC2D1dds)
  
  C1D1vC2D1res <- results(C1D1vC2D1dds, contrast = c("timepoint","C2D1", "C1D1"))
  C1D1vC2D1res <- lfcShrink(C1D1vC2D1dds, contrast = c("timepoint","C2D1", "C1D1"), res=C1D1vC2D1res, type="ashr")
  C1D1vC2D1res <- C1D1vC2D1res[!is.na(C1D1vC2D1res$padj),]
  
  
  #Extract Stats for Pathway Analysis
  BaselinevC1D1stats <- BaselinevC1D1res$log2FoldChange
  names(BaselinevC1D1stats) <- row.names(BaselinevC1D1res)

  BaselinevC2D1stats <- BaselinevC2D1res$log2FoldChange
  names(BaselinevC2D1stats) <- row.names(BaselinevC2D1res)
  
  C1D1vC2D1stats <- C1D1vC2D1res$log2FoldChange
  names(C1D1vC2D1stats) <- row.names(C1D1vC2D1res)
  
```

For each comparison, a datatable of results and volcanoplot will be provided. Plotted on the y-axis is the adjusted p-value, on the x-axis is the Log2 Fold Change.

## Baseline vs C1D1 DE 

Positive Log2 Fold Change = enriched in C1D1 samples.
Negative Log2 Fold Change = enriched in Baseline samples.

Saved are going to be the genes with an absolute Log2FoldChange > 0.5 and a Padj < 0.05

```{r}
##BaselinevC1D1
datatable(data.frame(BaselinevC1D1res))
#volcano plot
EnhancedVolcano(BaselinevC1D1res, lab=row.names(BaselinevC1D1res), 
                x='log2FoldChange', y='padj', FCcutoff = 0.5,
                pCutoff = 0.05, title = "Paired DE Results by Time -- BaselinevC1D1")

# significantly DE genes
BaselinevC1D1res_TopDE <- BaselinevC1D1res[BaselinevC1D1res$padj < 0.05 &
                              abs(BaselinevC1D1res$log2FoldChange) > 0.5,]
#order by logFC
BaselinevC1D1res_TopDE = BaselinevC1D1res_TopDE[order(BaselinevC1D1res_TopDE$log2FoldChange, decreasing = T),]

# the total number if significantly DE genes
cat("The number of significantly DE genes:\n",
nrow(BaselinevC1D1res_TopDE))

# count how many up- and down-regulated genes
cat("The number of down- and up-regulated genes:\n",
table(BaselinevC1D1res_TopDE$log2FoldChange >0))

write.csv(BaselinevC1D1res_TopDE, paste0(rootDir,"output_data/Baseline_vs_C1D1_TopDE.csv"))

```

## Baseline vs C2D1 DE 

Positive Log2 Fold Change = enriched in C2D1 samples.
Negative Log2 Fold Change = enriched in Baseline samples.

Saved are going to be the genes with an absolute Log2FoldChange > 0.5 and a Padj < 0.05

```{r}
##BaselinevC2D1
datatable(data.frame(BaselinevC2D1res))
# volcano
EnhancedVolcano(BaselinevC2D1res, lab=row.names(BaselinevC2D1res), 
                x='log2FoldChange', y='padj', FCcutoff = 0.5,
                pCutoff = 0.05, title = "Paired DE Results by Time -- BaselinevC2D1")

# significantly DE genes
BaselinevC2D1res_TopDE <- BaselinevC2D1res[BaselinevC2D1res$padj < 0.05 &
                              abs(BaselinevC2D1res$log2FoldChange) > 0.5,]
#order by logFC
BaselinevC2D1res_TopDE = BaselinevC2D1res_TopDE[order(BaselinevC2D1res_TopDE$log2FoldChange, decreasing = T),]

# the total number if significantly DE genes
cat("The number of significantly DE genes:\n",
nrow(BaselinevC2D1res_TopDE))

# count how many up- and down-regulated genes
cat("The number of down- and up-regulated genes:\n",
table(BaselinevC2D1res_TopDE$log2FoldChange >0))

write.csv(BaselinevC2D1res_TopDE, paste0(rootDir,"output_data/Baseline_vs_C2D1_TopDE.csv"))

```

## C1D1 vs C2D1 DE 

Positive Log2 Fold Change = enriched in C2D1 samples.
Negative Log2 Fold Change = enriched in C1D1 samples.

Saved are going to be the genes with an absolute Log2FoldChange > 0.5 and a Padj < 0.05

```{r}
##C1D1vC2D1
datatable(data.frame(C1D1vC2D1res))
# volcano plot
EnhancedVolcano(C1D1vC2D1res, lab=row.names(C1D1vC2D1res), 
                x='log2FoldChange', y='padj', FCcutoff = 0.5,
                pCutoffCol = 'padj', pCutoff = 0.05, title = "Paired DE Results by Time -- C1D1vC2D1")

# significantly DE genes
C1D1vC2D1res_TopDE <- C1D1vC2D1res[C1D1vC2D1res$padj < 0.05 &
                              abs(C1D1vC2D1res$log2FoldChange) > 0.5,]
#order by logFC
C1D1vC2D1res_TopDE = C1D1vC2D1res_TopDE[order(C1D1vC2D1res_TopDE$log2FoldChange, decreasing = T),]

# the total number if significantly DE genes
cat("The number of significantly DE genes:\n",
nrow(C1D1vC2D1res_TopDE))

# count how many up- and down-regulated genes
cat("The number of down- and up-regulated genes:\n",
table(C1D1vC2D1res_TopDE$log2FoldChange >0))

write.csv(C1D1vC2D1res_TopDE, paste0(rootDir,"output_data/C1D1_vs_C2D1_TopDE.csv"))

```

# Hallmark pathway analysis

Gene set statistics are run with [fgsea](https://www.biorxiv.org/content/10.1101/060012v3) with Hallmark pathways from [MSigDb](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3106198/).

## Baseline vs C1D1 

Positive Normalized Enrichment Score = enriched in C1D1 samples.
Negative Normalized Enrichment Score = enriched in Baseline samples.


```{r}
# Get Pathways
hallmark_df = msigdbr(species = "Homo sapiens", category = "H")
hallmark_list = hallmark_df %>% split(x = .$gene_symbol, f = .$gs_name)

# run fgsea
  BaselinevC1D1gsResults <- as.data.frame(fgsea(pathways=hallmark_list, stats=BaselinevC1D1stats))
#BaselinevC1D1
datatable(BaselinevC1D1gsResults)
fwrite(BaselinevC1D1gsResults, file=paste0(rootDir,"output_data/BaselinevC1D1_HALLMARK.csv"))

```

## Baseline vs C2D1 

Positive Normalized Enrichment Score = enriched in C2D1 samples.
Negative Normalized Enrichment Score = enriched in Baseline samples.

```{r}
  #Complete Pathway Analysis
  BaselinevC2D1gsResults <- as.data.frame(fgsea(pathways=hallmark_list, stats=BaselinevC2D1stats))
  
#BaselinevC2D1
datatable(BaselinevC2D1gsResults)
fwrite(BaselinevC2D1gsResults, file=paste0(rootDir,"output_data/BaselinevC2D1_HALLMARK.csv"))

```

## C1D1 vs C2D1 

Positive Normalized Enrichment Score = enriched in C2D1 samples.
Negative Normalized Enrichment Score = enriched in C1D1 samples.

```{r}
C1D1vC2D1gsResults <- as.data.frame(fgsea(pathways=hallmark_list, stats=C1D1vC2D1stats))
    
#C1D1vC2D1
datatable(C1D1vC2D1gsResults)
fwrite(C1D1vC2D1gsResults, file=paste0(rootDir,"output_data/C1D1vC2D1_HALLMARK.csv"))

```

## HALLMARK Pathway Dotplot
Create a dotplot with GSEA results for different comparisons. Plot significant pathways only (adjusted p-values < 0.05). Color is the normalized enrichment score. Red/blue means higher/lower expression on the second time point.


```{r}
# merge pathwyay analysis results for plotting
  pathRes = rbind(cbind(comparison = rep("Baseline_vs_C1D1",nrow(BaselinevC1D1gsResults)),BaselinevC1D1gsResults),
  cbind(comparison = rep("Baseline_vs_C2D1",nrow(BaselinevC2D1gsResults)),BaselinevC2D1gsResults),
  cbind(comparison = rep("C1D1_vs_C2D1",nrow(C1D1vC2D1gsResults)),C1D1vC2D1gsResults))

 # make NES for not sugnificant pathways as NA
  pathRes_ns = pathRes %>% mutate(NES, NES = ifelse(padj >0.05, NA, NES))
  
  # plot: dot plot
  p = ggplot(data = pathRes_ns %>% filter(!is.na(NES)), aes(x = comparison, y = pathway, color = NES)) + 
    geom_point(size = 4) +
  #  scale_color_gradient(low = "red", high = "blue") +
    scale_colour_gradient2(low = "blue", high = "red") +
    scale_x_discrete(labels = c("Baseline vs C1D1", "Baseline vs C2D1","C1D1 vs C2D1")) +
    theme(axis.text.x = element_text(angle = 90, vjust = 0.5))

  print(p)

```

```{r}
sessionInfo()
```